### DISCUSSION TITLE
Google Domains DNS challenge

### DISCUSSION THREAD

**Discussion by CapgrasDelusion2** [NginxProxyManager/nginx-proxy-manager#2646] [General]
Google Domains has finally created an API for DNS challenges for SSL certs. (Google Cloud has had it for a while I believe but not domains). 

[Relevant post from Google is here](https://security.googleblog.com/2023/03/google-trust-services-now-offers-tls.html) 

Any chance this could get implemented into the DNS challenge section of NPM, or does anyone know a way to do it manually through NPM? Otherwise I'll try certbot directly.

EDIT: certbot worked perfectly with the [certbot-dns-google-domains](https://github.com/aaomidi/certbot-dns-google-domains) plugin. Imported the cert and it works great. Native integration to NPM would be awesome if possible just to make renewal easier, but as is it's functional.

**Comment by iveney** [comment]
FWIW, I was about to ask and found it was actually integrated to the latest dev branch. I manually pulled the source and build the frontend, and replaced the ones in my docker container and it worked well. One can probably find the commit that contains the change and use the docker image generated with it. 

https://github.com/NginxProxyManager/nginx-proxy-manager/blob/develop/global/certbot-dns-plugins.js#L279-L287
